frame/elections-phragmen/src/lib.rs

```diff
@@ -904,14 +904,20 @@ impl<T: Trait> Module<T> {
 			to_votes(Self::locked_stake_of(who))
 		};
 
-		let voters_and_votes = Voting::<T>::iter()
-			.map(|(voter, (stake, targets))| { (voter, to_votes(stake), targets) })
+		// used for prime election.
+		let voters_and_stakes = Voting::<T>::iter()
+			.map(|(voter, (stake, targets))| { (voter, stake, targets) })
+			.collect::<Vec<_>>();
+		// used for phragmen.
+		let voters_and_votes = voters_and_stakes.iter()
```

Comment:
Re-iterating could be heavy. Would imply we need to re-do weights. Also, could we not just re-use `voters_and_stakes` perhaps convertingn the `stake` item back to its original type?

Reply:
> Re-iterating could be heavy. Would imply we need to re-do weights

Weight of this is already maximum of the block, can't get any worse. 

> Also, could we not just re-use voters_and_stakes perhaps converting the stake item back to its original type?

This is practically what we are doing. second iter is in-memory and won't go into storage again. `.cloned()` is inevitable because we need to pass `voters_and_votes` by value to `seq_phragmen`.